certify/dns_certify.ml

```diff
@@ -83,26 +126,35 @@ let pp_q_err ppf = function
   | `Unexpected_reply r -> Fmt.pf ppf "unexpected reply %a" Packet.pp_reply r
   | `No_tlsa -> Fmt.pf ppf "No TLSA record found"
 
+let tlsas_to_certchain public_key tlsas =
+  let certificates, ca_certificates =
+    Rr_map.Tlsa_set.fold (fun tlsa (certs, cacerts as acc) ->
+        if is_certificate tlsa || is_ca_certificate tlsa then
+          match X509.Certificate.decode_der tlsa.Tlsa.data with
+          | Ok cert ->
+            if is_certificate tlsa then
+              (cert :: certs, cacerts)
+            else (* must be is_ca_certificate *)
```

Comment:
does this comment suggest we should check for `is_ca_certificate` ?

Reply:
that's checked some lines above.. this code will only ever be executed if either `is_certificate` or `is_ca_certificate` is true.. checking it here again means I'd need to handle the impossible case of neither being true, which I'd like to avoid (but the code may be refactored for good to avoid these cases)

Reply:
I think it'd be good if the comment could explain that (I was a bit unsure if `must be` meant "we can logically derive that it is because we just checked" or "it really really should be, but we don't know if it is")